### PR TITLE
Use module-global `Metrics` instead of `this` in metrics.js

### LINKED ISSUE #1135: variant error: this.experimentFeaturesChanged is not a function
I'm unable to send variant data from universal search to test pilot.

[Here](https://github.com/mozilla/universal-search/pull/261/commits/89f536f346a24dd7f1d17143c7714f89a3ee858e) is a universal search commit that mostly contains variant-related code.

Each time I send the data over, I see this error in the browser toolbox console:

```
TypeError: this.experimentFeaturesChanged is not a function
Stack trace:
module.exports.onReceiveVariantDefs@resource://gre/modules/commonjs/toolkit/loader.js -> resource://testpilot-addon/lib/metrics.js:163:5
Observer<.observe@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/system/events.js:81:7
Search.prototype.fetchVariantData/sendPing@chrome://universalsearch-lib/content/universal-search.js:109:7
setTimeout_timer@resource://gre/modules/Timer.jsm:30:5
```

Looks like the value of `this` isn't properly bound at line 163 of lib/metrics.js.

### PR DESCRIPTION
**Pull request by lmorchard**
Fixes #1135

### CODE REVIEW

File: addon/lib/metrics.js [hunk 1]
```diff
@@ -75,8 +75,8 @@ module.exports = {
 
     this.maybePingTelemetry();
 
-    Events.on(EVENT_SEND_METRIC, this.onExperimentPing);
-    Events.on(EVENT_RECEIVE_VARIANT_DEFS, this.onReceiveVariantDefs);
+    Events.on(EVENT_SEND_METRIC, this.onExperimentPing.bind(this));
+    Events.on(EVENT_RECEIVE_VARIANT_DEFS, this.onReceiveVariantDefs.bind(this));
```
Comment (jaredhirsch) [inline addon/lib/metrics.js]: This won't work because we use `Events.off` to disconnect the functions further down

Comment (jaredhirsch) [inline addon/lib/metrics.js]: You'll need to assign the bound functions somewhere and pass 'em to `on` and `off`

**Comment by lmorchard** [comment]
Updated to something more drastic: Switched away from `this` to using a `Metrics` global reference to `module.exports` - [we use that pattern elsewhere](https://github.com/mozilla/testpilot/blob/master/addon/lib/toolbar-button.js#L89) and `module.exports` is a weird object anyway. (e.g. it's frozen such that any assignments to `this.foo = 'bar'` is a runtime error, so we shouldn't treat it like a typical object)

**Comment by jaredhirsch** [comment]
lgtm but I don't have time right now to verify locally, I can look later this afternoon

**Comment by jaredhirsch** [comment]
looks good, merging

**Pull request merged**